lib/StripeResource.js

```diff
@@ -403,21 +407,23 @@ StripeResource.prototype = {
 
       const apiVersion = this._stripe.getApiField('version');
       requestData = data;
-      const headers = this._defaultHeaders(
-        auth,
-        requestData.length,
-        apiVersion
-      );
-
-      this._stripe.getClientUserAgent((cua) => {
-        headers['X-Stripe-Client-User-Agent'] = cua;
-
-        if (options.headers) {
-          Object.assign(headers, options.headers);
-        }
-
-        this._addTelemetryHeader(headers);
 
+      this._stripe.getClientUserAgent((clientUserAgent) => {
+        // If this is a POST and we allow multiple retries, set a idempotency key if one is not
+        // already provided.
+        const defaultIdempotencyKey =
```

Comment:
Any reason why we don't do this in `_makeHeaders`, and only if a user supplied idempotency key is not passed in?

Reply:
Made this a bit nicer I think...